packages/live-preview-sdk/src/inspectorMode/utils.ts

```diff
@@ -53,21 +58,66 @@ export function getInspectorModeAttributes(
 /**
  * Query the document for all tagged elements
  */
-export function getAllTaggedElements(root = window.document): Element[] {
+export function getAllTaggedElements(root = window.document, ignoreManual?: boolean): Element[] {
   // The fastest way to look up & iterate over DOM. Ref:
   // https://stackoverflow.com/a/2579869
   //
   // Terminology:
   // FILTER_SKIP: Skip the current node
   // FILTER_REJECT: Skip the current node and all its children
   const walker = document.createTreeWalker(root, NodeFilter.SHOW_ALL, (node) => {
+    if (node.nodeType === Node.TEXT_NODE) {
+      const text = node.textContent ?? '';
+
+      const { origin } = decode(text) ?? {};
+
+      if (origin !== 'contentful.com') {
+        return NodeFilter.FILTER_SKIP;
+      }
+
+      return isTaggedElement(node.parentElement)
+        ? ignoreManual
+          ? NodeFilter.FILTER_ACCEPT
+          : NodeFilter.FILTER_SKIP
+        : NodeFilter.FILTER_ACCEPT;
+    }
+
+    if (ignoreManual) {
+      return NodeFilter.FILTER_SKIP;
+    }
+
     return isTaggedElement(node) ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_SKIP;
   });
 
   const elements: Element[] = [];
 
   while (walker.nextNode()) {
-    elements.push(walker.currentNode as Element);
+    const node = walker.currentNode;
+
+    if (node.nodeType === Node.ELEMENT_NODE) {
+      elements.push(walker.currentNode as Element);
+      continue;
+    }
+
+    // Handle Encoded strings
+    const { cf } = decode(node.textContent ?? '') as SourceMapMetadata;
```

Comment:
> as SourceMapMetadata

Is this not enforced by the return type of the function?

Seems unnecessary to decode an empty string
```
if (node.textContent) {
  const { cf } = decode(node.textContent)
  ... set attributes
}
```

Reply:
> Is this not enforced by the return type of the function?

It can be undefined with types alone. I'm forcing it it because we already ensure it exists in the preview code block